File: src/test/java/com/raychenon/leetcode/CountSortedVowelStringsTest.java
package com.raychenon.leetcode;

import junit.framework.TestCase;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * User: raychenon
 * Date: 19/1/2021
 * Time: 11:21 PM
 * https://leetcode.com/problems/count-sorted-vowel-strings/
 */
public class CountSortedVowelStringsTest {

    private static CountSortedVowelStrings instance;

    @BeforeClass
    public static void setUp() {
        instance = new CountSortedVowelStrings();
    }

    @Test
    public void example1Test() {
        assertCountSortedVowel(5, 1);
    }

    @Test
    public void example2Test() {
        assertCountSortedVowel(15, 2);
    }

    @Test
    public void example3Test() {
        assertCountSortedVowel(66045, 33);
    }

    void assertCountSortedVowel(int expected, int n) {
        TestCase.assertEquals(expected, instance.countVowelStringsDP(n));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/CountSortedVowelStringsTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 21/1/2021
 * https://leetcode.com/problems/count-sorted-vowel-strings/
 */
class CountSortedVowelStringsTest {

    @Test
    fun example1Test() {
        assertCountSortedVowel(5, 1)
    }

    @Test
    fun example2Test() {
        assertCountSortedVowel(15, 2)
    }

    @Test
    fun example3Test() {
        assertCountSortedVowel(66045, 33)
    }

    fun assertCountSortedVowel(expected: Int, n: Int) {
        TestCase.assertEquals(expected, CountSortedVowelStrings.countVowelStringsDPBottomUp(n))
        TestCase.assertEquals(expected, CountSortedVowelStrings.countVowelStringsDPBottomUp1D(n))
    }
}